### PR TITLE
Add a "showErrorOutput" property

### PR DESCRIPTION
**Pull request by mindhog**
Add a property to let us dump test output and final status in real-time to the
console.

---
This change is [Reviewable](https://reviewable.io/reviews/google/nomulus/237)

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -3,5 +3,6 @@ pluginsUrl=
 uploaderDestination=
 uploaderCredentialsFile=
 uploaderMultithreadedUpload=
+verboseTestOutput=
```
Comment (gbrodman) [inline gradle.properties]: Should we make this explicitly false?

File: build.gradle [hunk 2]
```diff
@@ -256,6 +259,31 @@ subprojects {
   }
 }
 
+// If "-P verboseTestOutput=true" is passed in, configure all subprojects to dump all of their
+// output and final test status (pass/fail, errors) for each test class.
+if (verboseTestOutput) {
+  gradle.projectsEvaluated({
+    subprojects {
```
Comment (gbrodman) [inline build.gradle]: What's the reasoning for putting this here instead of in the subprojects block above?

**Comment by gbrodman** [comment]
Is the thinking here that it'll be easier to debug remote build failures if the exceptions are printed to the console, rather than having to dig for remote files?

File: gradle.properties (hunk 1 above)
Comment (mindhog) [reply gradle.properties]: That actually doesn't work, I think gradle is just treating the property as a string and will treat it as false in a conditional only if it's empty.

File: build.gradle (hunk 2 above)
Comment (mindhog) [reply build.gradle]: Yeah, I guess that merits a comment, which I've added.  The short answer is we have to apply these to all of the test targets, and they aren't defined when the main subprojects section is run.

**Comment by mindhog** [comment]
> Is the thinking here that it'll be easier to debug remote build failures if the exceptions are printed to the console, rather than having to dig for remote files?

On the contrary, I added it for local builds.  I don't know how it is in an IDE, but on the shell it's a PITA to copy the link, switch over to the browser and dig down into the test failure.  Also, having test results scroll out in real-time means I can get to work on a fix faster and greatly reduces my cycle times.

**Comment by mindhog** [comment]
> I see -- that's a good reason.
> 
> > _[Reviewable](https://reviewable.io/reviews/google/nomulus/237#-:-LmtbtdM6M8QPVYBrr2T:bdtl5wk)_ status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @mindhog)
> 
> _[build.gradle, line 266 at r1](https://reviewable.io/reviews/google/nomulus/237#-Lmocg3rBIol7erYXweZ-r1-266:-Lmtbs7CCU8Eyd5njbkG:b-u5vc8w) ([raw file](https://github.com/google/nomulus/blob/75bce00a5002a65265022a813e1bf5fbec04ecb8/build.gradle#L266)):_
> 
> Previously, mindhog (Michael Muller) wrote…
> Sounds good, thanks for adding that
> 
> _[gradle.properties, line 6 at r1](https://reviewable.io/reviews/google/nomulus/237#-Lmocg3rBIol7erYXwe_-r1-6:-LmtWdU21GZDNC8pa5n4:b-u5htll) ([raw file](https://github.com/google/nomulus/blob/75bce00a5002a65265022a813e1bf5fbec04ecb8/gradle.properties#L6)):_
> 
> Previously, mindhog (Michael Muller) wrote…
> Looking at how we treat enableDependencyLocking, we'd have to say "if (verboseTestOutput.toBoolean()) {.."
> 
> Thoughts? I think either probably works

We should probably be consistent with enableDependencyLocking.  Done.

**Pull request merged**